src/validator/RecoveryPlugin.sol

```diff
@@ -0,0 +1,202 @@
+        address oldOwner = recoveryPluginStorage[msg.sender].owner;
+        recoveryPluginStorage[msg.sender].owner = _newOwner;
+        for (uint256 i = 0; i < guardians[msg.sender].length; i++) {
+            guardians[msg.sender][i].approved = false;
+        }
+        emit OwnerChanged(msg.sender, oldOwner, _newOwner);
+    }
+
+    function addGuardian(
+        Guardian[] calldata _guardians,
+        uint256 _thresholdWeight,
+        uint256 delay
+    ) external {
+        for (uint256 i = 0; i < _guardians.length; i++) {
+            if (_guardians[i].weight <= 0) {
+                revert();
+            }
+            if (_guardians[i].guardian == address(0)) {
+                revert();
+            }
+            if (_guardians[i].guardian == msg.sender) {
+                revert();
+            }
+            if (_guardians[i].approved) {
+                revert();
+            }
+            guardians[msg.sender].push(_guardians[i]);
+        }
+        thresholdWeight[msg.sender] = _thresholdWeight;
+        recoveryDelay[msg.sender] = block.timestamp + delay;
+        emit GuardianAdded(msg.sender, _guardians, _thresholdWeight);
+    }
+
+    function verifyGuardians(
+        bytes32 hash,
+        bytes[] memory signatures
+    ) internal returns (uint256) {
+        uint256 weight = 0;
+        for (uint256 i = 0; i < signatures.length; i++) {
+            if (signatures[i].length != 65) {
+                revert();
+            } else {
+                if (
+                    validateGuardianSignature(
+                        hash,
+                        signatures[i],
+                        guardians[msg.sender][i].guardian
+                    ) ==
+                    0 &&
+                    !guardians[msg.sender][i].approved
+                ) {
+                    unchecked {
+                        weight += guardians[msg.sender][i].weight;
+                    }
+                }
+            }
+        }
+        return weight;
+    }
+
+    function initRecovery(
+        address _newOwner,
+        bytes32 hash,
+        bytes[] memory signatures
+    ) internal {
+        require(
+            _newOwner != address(0),
+            "RecoveryPlugin: new owner is zero address"
+        );
+        require(hash != bytes32(0), "RecoveryPlugin: hash is zero");
+        require(
+            block.timestamp >= recoveryDelay[msg.sender],
+            "RecoveryPlugin: recovery delay not reached"
+        );
+        uint256 weight = verifyGuardians(hash, signatures);
+        require(
+            weight >= thresholdWeight[msg.sender],
+            "RecoveryPlugin: weight is not enough"
+        );
+        address oldOwner = recoveryPluginStorage[msg.sender].owner;
+        require(
+            oldOwner != _newOwner,
+            "RecoveryPlugin: new owner is the same as old owner"
+        );
+        changeOwner(_newOwner);
+    }
+
+    function validateUserOp(
```

Comment:
Since enable can only be called after passing the validateUserOp(), I think we should do some more tricks that will enable guardians to call `enable()` function.

maybe adding a mode will do?